### PR TITLE
Feat: add definitions to addon detail API, fix addon cache, async download files

### PR DESCRIPTION
**Pull request by chivalryq**
### Description of your changes

1. Add definitions to addon detail API, each definition contains `name`,`type`,`description` fields.
2. optimize addon cache usage
3. download the addon's files parallelly
image

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/apiserver/rest/apis/v1/types.go [hunk 1]
```diff
@@ -97,7 +97,16 @@ type DetailAddonResponse struct {
 	UISchema  []*utils.UIParameter `json:"uiSchema"`
 
 	// More details about the addon, e.g. README
-	Detail string `json:"detail,omitempty"`
+	Detail      string `json:"detail,omitempty"`
+	Definitions []*AddonDefinition
```
Comment (hongchaodeng) [inline pkg/apiserver/rest/apis/v1/types.go:101]: json tag

File: pkg/apiserver/rest/apis/v1/types.go [hunk 2]
```diff
@@ -97,7 +97,16 @@ type DetailAddonResponse struct {
...
+}
+
+// AddonDefinition is definition an addon can provide
+type AddonDefinition struct {
+	Name string `json:"name,omitempty"`
+	// can be component/trait...definition
+	DefType     string `json:"def_type,omitempty"`
```
Comment (hongchaodeng) [inline pkg/apiserver/rest/apis/v1/types.go:108]:
```suggestion
	DefType     string `json:"type,omitempty"`
```

File: pkg/apiserver/rest/apis/v1/types.go (hunk 1 above)
Comment (chivalryq) [reply pkg/apiserver/rest/apis/v1/types.go]: done

File: pkg/apiserver/rest/apis/v1/types.go (hunk 2 above)
Comment (chivalryq) [reply pkg/apiserver/rest/apis/v1/types.go]: done

**Review by hongchaodeng — APPROVED** [review]
lgtm
please fix CI

**Comment by chivalryq** [comment]
CI fail because some error occured on our CI machine ...

**Review by barnettZQG — APPROVED** [review]

**Pull request merged**